### PR TITLE
Add test cases for the file metric calculation for C++

### LINKED ISSUE #84: Add tests for file metrics for C++
Add test cases for C++ for all supported file metrics

### PR DESCRIPTION
**Pull request by ResistantBear**
Adds test source and header files in the C++ programming languages. These files are tested to compile correctly using the LLVM C++ compiler clang that comes with MacOS (command `clang -c -xc++ -std=c++20 <file>`)
Adds test cases that check if the file metrics are calculated correctly if these C++ files are used as input. 

Additional notes:
- Fixes that C++ for-each-loops (syntax node `for_range_loop`) were not counted when calculating the complexity metric

Resolves #84

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,7 +58,7 @@ This is quite slow and can take up to one or two hours but can provide good resu
 Counts expressions that branch the control flow (if-statements, loops, catch-blocks, etc. - but no else/default/finally statements) and other expressions that are considered to increase the complexity of the code inside a file:
 
 -   function declarations (including lambda expressions)
--   logical operations (like AND and OR)
+-   binary logical operations (like AND and OR)
```
Comment (ResistantBear) [inline README.md:61]: Clearification, as we do not count unary logical operations, like the use of the `!` / negation operator.

File: test/parser/TestHelper.ts [hunk 2]
```diff
@@ -56,7 +56,7 @@ export function sortCouplingResults(couplingResult: CouplingResult) {
  * @param metric Name of the metric.
  * @param expected Expected test result.
  * */
-export async function testFileMetrics(inputPath: string, metric: FileMetric, expected: number) {
+export async function testFileMetric(inputPath: string, metric: FileMetric, expected: number) {
```
Comment (ResistantBear) [inline test/parser/TestHelper.ts:59]: renamed `testFileMetrics` to `testFileMetric`, as it only checks one value.

File: test/parser/UnknownFiles.test.ts [hunk 3]
```diff
@@ -1,7 +1,5 @@
-import { getFileMetrics, testFileMetrics } from "./TestHelper";
-import { FileMetric } from "../../src/parser/metrics/Metric";
+import { getFileMetrics } from "./TestHelper";
 import fs from "fs";
-import { GenericParser } from "../../src/parser/GenericParser";
```
Comment (ResistantBear) [inline test/parser/UnknownFiles.test.ts:4]: Removed unused imports

**Review by michaelalper-cezik-mw — APPROVED** [review]
very nice example source files, everything looks good

**Pull request merged**